lightning_pose/data/datasets.py

```diff
@@ -328,10 +327,156 @@ def __getitem__(self, idx: int) -> HeatmapLabeledExampleDict:
 
         """
         example_dict: BaseLabeledExampleDict = super().__getitem__(idx)
-        if len(self.imgaug_transform) == 1 and isinstance(self.imgaug_transform[0], iaa.Resize):
-            # we have a deterministic resizing augmentation; use precomputed heatmaps
-            example_dict["heatmaps"] = self.label_heatmaps[idx]
+        example_dict["heatmaps"] = self.compute_heatmap(example_dict)
+        return example_dict
+
+
+class DetectorDataset(BaseTrackingDataset):
```

Comment:
cleaner if you inherit from `HeatmapDataset` - then you don't need to to redefine `compute_heatmap`, `output_shape`, etc., as well as some of the attribute setting in `__init__` (`output_sigma`, `uniform_heatmaps`, etc.).

you can still overwrite `__getitem__` to do whatever you want